### DISCUSSION TITLE
Is it possible to reference the status of a checkbox in a standard text (rich text) note with javascript?

### DISCUSSION THREAD

**Discussion by DanielBurdock** [TriliumNext/Trilium#4480] [Q&A] [answered]
Apologies if this is a bit of a basic question, but I was just wondering if this was possible and how to go about doing it if so. I would preferably like to make a checkbox list within a standard text note and then generate a new note based on the status of those checkboxes that pulls from the text next to it.

The general idea that I have is to make an inventory of food I have in so I can then generate a shopping list based on what I've run out of. So for example:

- [ ] Black pepper
- [ ] Garlic powder
- [x] Sea salt

Would generate a new note 'shopping list' after a setting up a button to press:

- Black pepper
- Garlic powder

Just wanted to check if this is possible. If not I suppose I could try and do it with HTML or attributes but I think this method would be preferable.

**Comment by zerebos** [comment, marked as answer]
With plain JS you can check for the `checked` DOM attribute:
![image](https://github.com/zadam/trilium/assets/6865942/8134e57a-7136-4c26-b267-3f63dbd4a88b)

Or just checked the state directly:
![image](https://github.com/zadam/trilium/assets/6865942/6f0a10f3-1123-41da-b97f-ad50a6174cdb)

**Reply by DanielBurdock** [reply]
Thanks a lot :) I've not quite got the hang of implementing it yet but I'm sure I'll work it out soon. Really appreciate that!
